semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -127,7 +130,12 @@ let generic_to_json env (key : key) ast =
 
 let read_string_wrap e =
   match e with
-  | G.L (String (value, t)) -> Some (value, t)
+  | G.L (String (value, t)) ->
+      (* BUG: need to unescape the string s e.g. {|\"x\"|} -> {|"x"|}
```

Comment:
same, the bug is not here.

The bug is in Json_to_generic.program which should take an option to request to unescape the string, because we are in the context of parsing a rule in which case we want all the strings to get their "final" value.

Reply:
These comments are meant to indicate that when we add a field containing the unescaped string, that's the one we should use here.

Reply:
I updated the comments.